src/components/FileSidebar.tsx

```diff
@@ -10,14 +10,17 @@ interface FileSidebarProps {
   currentFile: FileName
   onFileSelect: (filename: FileName) => void
   className?: string
+  fileSidebarState: ReturnType<typeof useState<boolean>>
 }
 
 const FileSidebar: React.FC<FileSidebarProps> = ({
   files,
   currentFile,
   onFileSelect,
   className,
+  fileSidebarState,
```

Comment:
huuuuge antipattern

always a different way to do this

Reply:
separating into two vars w/ conventional names is probably best

- `sidebarOpen`
- `onToggleSidebar`